docs/madr/2-sqlite.md

```diff
@@ -0,0 +1,100 @@
+# Storing ANT Registry State in AO Process
+
+- Status: proposed
+- Approvers: [Dylan], [Ariel], [Phil]
+- Date: 2024-07-22
+- Authors: [Atticus]
+
+## Context and Problem Statement
+
+There are multiple options for storing state in ao Lua processes, namely thru
+the use of the Lua `table` primitive, and a special build called aos-sqlite
+which exposes the lua-sqlite3 library for using in-memory databases.
+
+Directly using Lua tables for complex data has shown a need for better tooling
+in manipulating and searching data, making the SQLite implementation a potential
+solution. However, its novelty raises concerns about reliability in larger
+products.
+
+## Decision Drivers
+
+The main drivers for this decision are:
+
+- **Scalability**
+  - The current method of using tables has scalability issues that SQLite can
+    potentially address.
+- **Performance**
+  - Improved performance in handling larger datasets with SQLite.
+- **Risk Management**
+  - Mitigating risks in larger products by testing the module in a controlled
+    environment first.
+- **Ecosystem Integration**
+  - Exploring new modules and their potential benefits within the ao ecosystem.
+- **Maintenance**
+  - How many development cycles will be needed to maintain the solution.
+
+## Considered Options
+
+### Option 1: Continue Using Existing Tables
+
+- **Pros:**
+  - Known and stable.
+  - No need for additional testing or integration.
+  - Smaller storage requirement than sqlite
+- **Cons:**
+  - Demonstrated scalability issues when attempting to retrieve 150k records in
+    ANT's.
+  - Limited performance with larger datasets, namely deeply nested tables.
```

Comment:
assuming we contstruct our tables efficiently, we can get around most of this. i think the major limitation is more advanced schemas require a lot more mental energy to efficiently search/query without the optimizations of SQL queries provided by sqlite